aws/resource_aws_network_acl_rule_test.go

```diff
@@ -632,15 +643,13 @@ resource "aws_network_acl" "test" {
 }
 
 resource "aws_network_acl_rule" "test" {
-  from_port       = -1
   icmp_code       = -1
   icmp_type       = -1
   ipv6_cidr_block = "::/0"
   network_acl_id  = "${aws_network_acl.test.id}"
-  protocol        = 58
+  protocol        = "ipv6-icmp"
```

Comment:
updated here to the string value b/c if the protocol is sent as an int, while the resource ID will be generated with the value provided as input https://github.com/terraform-providers/terraform-provider-aws/blob/c8027b65401471582d1d699b8638efc164840b76/aws/resource_aws_network_acl_rule.go#L167 , the protocol value that gets saved to the resource's state is the string identifier https://github.com/terraform-providers/terraform-provider-aws/blob/c8027b65401471582d1d699b8638efc164840b76/aws/resource_aws_network_acl_rule.go#L227 which then causes difficulty at `ImportState` time b/c `testAccAWSNetworkAclRuleImportStateIdFunc` will return the string value of the protocol and this will inevitably generate the wrong ID

Reply:
since we depend on `testAccAWSNetworkAclRuleImportStateIdFunc` to generate the ID for importing, is a better alternative to fix the output there instead of adjusting the resource params?

Reply:
Ideally the `ImportStateIdFunc` should handle any quirky behavior caused by multiple potential values for an argument, rather than forcing the test configurations to be updated. 👍